src/parser.js

```diff
@@ -667,11 +675,60 @@ function mapBlockStatement(node, meta, factory = b.blockStatement) {
 }
 
 function mapConditionalAssignment({variable, value, context, operator}, meta) {
-  let memberExpression = b.identifier(variable.base.value);
+  const {condition, memberExpression} = mapSoakedChainToConditions(variable);
+
+  return b.ifStatement(
+    condition,
+    b.blockStatement(
+      [
+        b.expressionStatement(
+          b.assignmentExpression(context || operator || '=', memberExpression, mapExpression(value, meta))
+        ),
+      ]
+    )
+  );
+}
+
+function mapConditionalUpdateOperator({first: value, context, operator, flip}) {
+  const {condition, memberExpression} = mapSoakedChainToConditions(value);
+
+  return b.ifStatement(
+    condition,
+    b.blockStatement(
+      [
+        b.expressionStatement(
+          b.updateExpression(context || operator, memberExpression, !flip)
+        ),
+      ]
+    )
+  );
+}
+
+function mapSoakedChainToConditions({properties, base}) {
+  let memberExpression = b.identifier(base.value || base.args[0].base.value);
   let condition;
 
-  for (const property of variable.properties) {
-    const identifier = b.identifier(property.name.value);
+  for (const property of properties) {
+    let identifier;
+    let computed = false;
+
+    if (property.constructor.name === 'Access') {
+      identifier = b.identifier(property.name.value);
+    } else {
+      if (property.index.properties.length > 0) {
```

Comment:
I might just bump this to be `else if` in the above contiional (and the last part being the `else`), the only thing shared is that it indicates that the thing is computed which is fine to put in both (or to just set equal to `property.constructor.name !== 'Access'` outside of this entirely).